### PR TITLE
Group page UI

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4990

## Description of Changes
- Added community group dashboard UI, accessible from this link `/:scope/members?tab=groups`.
- Added filter behaviors
- Added checks for admin-level features

## "How We Fixed It"
Not a bug

## Test Plan
Since this is a new feature we have to look for any things that seem a bit off

- Visit the `/:scope/members?tab=groups` page for any community, and verify that all elements work correctly
- Visit the same page with admin account and verify you see the create group button (this is only visible to admins)
- Verify the filters group (search filter will filter groups by name and dropdown filter will filter by joined status i.e if you have joined that group or not)

## Deployment Plan
N/A

## Other Considerations
- Didn't connect the API (its still in dev).

### CODE REVIEW

**Comment by mzparacha** [comment]
Ready for eng review. cc: @jnaviask @masvelio

File: packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx:137]: 'index' is defined but never used

**Review by masvelio — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/Members/GroupsSection/GroupsSection.tsx:137]: Forgot to use it as a `key`, will add that.

**Review by rbennettcw — APPROVED** [review]
Code looks good. Looks like it's using the query param as a feature flag.

Just want to confirm that there's no way a user can navigate to the page without the query param?

**Comment by mzparacha** [comment]
> Code looks good. Looks like it's using the query param as a feature flag.
> Just want to confirm that there's no way a user can navigate to the page without the query param?

Right now, yes only the flag in URL will show the page

**Comment by mzparacha** [comment]
### Update

The api is connected API.

Pending: need to confirm point 1 from [this](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769116790) comment.

**Comment by mzparacha** [comment]
### Update

The api is connected API.

Pending: need to confirm point 1 from [this](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769116790) comment.

**Comment by mzparacha** [comment]
### Update

The api is connected API.

Pending: need to confirm point 1 from [this](https://github.com/hicommonwealth/commonwealth/pull/5045#issuecomment-1769116790) comment.

**Comment by mzparacha** [comment]
Update in https://github.com/hicommonwealth/commonwealth/pull/5402, we added a new param to get the topics array. Updating the code according

**Comment by mzparacha** [comment]
The topics array is also displayed, along with some other minor changes.

**Comment by mzparacha** [comment]
### Status

1. The `GET /groups` route does return the memberships, but it returns an `address_id` in response. I couldn't find this in frontend, asked a question [here](https://commonxyz.slack.com/archives/C050AE0URFH/p1697814171186819) and awaiting response on that.
2. The filters will be done in https://github.com/hicommonwealth/commonwealth/issues/4989

We can merge this after 1 is sorted

cc: @jnaviask @CowMuon

**Comment by mzparacha** [comment]
The point 1 from [this](https://github.com/hicommonwealth/commonwealth/pull/5235#issuecomment-1772926663) comment is also sorted out. 

We can merge this PR (it also requires https://github.com/hicommonwealth/commonwealth/pull/5402 to be merged in)

cc: @jnaviask @CowMuon

**Comment by mzparacha** [comment]
The CI errors are fixed in https://github.com/hicommonwealth/commonwealth/pull/5275. Once we merge that to master, and merge master to this PR, we won't have CI errors here.

**Review by kurtisassad — APPROVED** [review]
LGTM after code quality fixes.

**Comment by zakhap** [comment]
Screen Shot 2023-10-25 at 4 02 01 PM
Screen Shot 2023-10-25 at 4 02 10 PM

When doing my review of QA, I noticed these.

**Comment by mzparacha** [comment]
As discussed on slack, this issue is only on frack (where gating is enabled) and not reproducible on local. I will create a bug ticket to fix this.

**Pull request merged**
